core/src/main/java/org/elasticsearch/action/ingest/GetPipelineResponse.java

```diff
@@ -24,25 +24,25 @@
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.xcontent.StatusToXContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.ingest.PipelineDefinition;
+import org.elasticsearch.ingest.IngestMetadata.PipelineConfiguration;
 import org.elasticsearch.rest.RestStatus;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public class GetPipelineResponse extends ActionResponse implements StatusToXContent {
 
-    private List<PipelineDefinition> pipelines;
+    private Set<PipelineConfiguration> pipelines;
 
     public GetPipelineResponse() {
     }
 
-    public GetPipelineResponse(List<PipelineDefinition> pipelines) {
+    public GetPipelineResponse(Set<PipelineConfiguration> pipelines) {
         this.pipelines = pipelines;
     }
 
-    public List<PipelineDefinition> pipelines() {
+    public Set<PipelineConfiguration> pipelines() {
```

Comment:
I was thinking about if duplicated ids were requested, but lets return a list. It would make the unit tests for this class a bit easier too.